mempool/bench_test.go

```diff
@@ -1,27 +1,75 @@
 package mempool
 
 import (
+	"crypto/rand"
 	"encoding/binary"
+	"math/big"
+	"strconv"
 	"testing"
 
 	"github.com/tendermint/tendermint/abci/example/kvstore"
 	cfg "github.com/tendermint/tendermint/config"
 	"github.com/tendermint/tendermint/proxy"
+	"github.com/tendermint/tendermint/types"
 )
 
-func BenchmarkCheckTx(b *testing.B) {
+// BenchmarkClistCheckTx-8   	10842198	        94.9 ns/op
+func BenchmarkClistCheckTx(b *testing.B) {
 	app := kvstore.NewApplication()
 	cc := proxy.NewLegacyLocalClientCreator(app)
 	mempool, cleanup := newLegacyMempoolWithAppAndConfig(cc, cfg.ResetTestRoot("mempool_test"), enumclistmempool)
 	defer cleanup()
+	for i := 0; i < b.N; i++ {
+		tx := make([]byte, 8)
+		binary.BigEndian.PutUint64(tx, uint64(i))
+		mempool.CheckTx(tx, nil, TxInfo{})
+	}
+}
 
+// BenchmarkLLRBCheckTx-8   	10889796	        99.2 ns/op
+func BenchmarkLLRBCheckTx(b *testing.B) {
+	app := kvstore.NewApplication()
+	cc := proxy.NewLegacyLocalClientCreator(app)
+	mempool, cleanup := newLegacyMempoolWithAppAndConfig(cc, cfg.ResetTestRoot("mempool_test"), enumllrbmempool)
+	defer cleanup()
 	for i := 0; i < b.N; i++ {
 		tx := make([]byte, 8)
 		binary.BigEndian.PutUint64(tx, uint64(i))
 		mempool.CheckTx(tx, nil, TxInfo{})
 	}
 }
 
+// BenchmarkClistRemoveTx-8   	  480782	      2151 ns/op
+func BenchmarkClistRemoveTx(b *testing.B) {
+	benchmarkRemoveTx(b, enumclistmempool)
+}
+
+// BenchmarkLLRBRemoveTx-8   	  541332	      2115 ns/op
+func BenchmarkLLRBRemoveTx(b *testing.B) {
+	benchmarkRemoveTx(b, enumllrbmempool)
+}
+
+func benchmarkRemoveTx(b *testing.B, enum mpEnum) {
+	app := kvstore.NewApplication()
+	cc := proxy.NewLegacyLocalClientCreator(app)
+	mempool, cleanup := newLegacyMempoolWithAppAndConfig(cc, cfg.ResetTestRoot("mempool_test"), enum)
+	defer cleanup()
+	size := 100000
+	txs := types.Txs{}
+	for j := 0; j < size; j++ {
+		rand, _ := rand.Int(rand.Reader, big.NewInt(100))
+		tx := "k" + strconv.Itoa(j) + "=v" + strconv.Itoa(j) + ",f," + strconv.FormatInt(rand.Int64(), 10)
+		txBytes := []byte(tx)
+		txs = append(txs, txBytes)
+		mempool.CheckTx(txBytes, nil, TxInfo{})
+	}
+
+	for i := 0; i < b.N; i++ {
```

Comment:
this is the part we actually want to test. you probably want to reset timer before remove